sr_data_visualization/scripts/sr_data_visualization/sr_data_visualizer_gui.py

```diff
@@ -0,0 +1,1029 @@
+        loadUi(ui_file, self._widget)
+        if __name__ != "__main__":
+            context.add_widget(self._widget)
+        self._widget.setWindowTitle("Dexterous Hand Data Visualizer")
+
+        # Set white background color
+        p = self._widget.palette()
+        p.setColor(self._widget.backgroundRole(), Qt.white)
+        self._widget.setPalette(p)
+        self.tab_widget_main = self._widget.findChild(QTabWidget, "tabWidget_main")
+        self.tabWidget_motor_stats = self._widget.findChild(QTabWidget, "tabWidget_motor_stats")
+
+        # Change tabs background color
+        p = self.tab_widget_main.palette()
+        stylesheet = """
+            QTabWidget>QWidget>QWidget{background: white;}
+            """
+        p.setColor(self.tab_widget_main.backgroundRole(), Qt.white)
+        self.tab_widget_main.setStyleSheet(stylesheet)
+
+        self.tab_widget_main.currentChanged.connect(self.tab_change)
+        self.tabWidget_motor_stats.currentChanged.connect(self.tab_change_mstat)
+
+        motor_stat_keys_file = os.path.join(rospkg.RosPack().get_path('sr_data_visualization'), 'config',
+                                            'data_visualiser_motor_stat_keys.yaml')
+        parameters_file = os.path.join(rospkg.RosPack().get_path('sr_data_visualization'), 'config',
+                                       'data_visualiser_parameters.yaml')
+
+        self.font_offset = -3
+
+        self._widget.resizeEvent = self.on_resize_main
+
+        with open(motor_stat_keys_file, 'r') as stream:
+            try:
+                self.motor_stat_keys = yaml.load(stream)
+            except yaml.YAMLError as exc:
+                print(exc)
+
+        with open(parameters_file, 'r') as stream:
+            try:
+                data_loaded = yaml.load(stream)
+                self._initialize(data_loaded)
+            except yaml.YAMLError as exc:
+                print(exc)
+
+        self._include_tactile_plugin()
+
+        self._setup_radio_buttons()
+
+        self.tabWidget_motor_stats.setCurrentIndex(0)
+        self.tabWidget_motor_stats.setCurrentIndex(1)
+        self.tabWidget_motor_stats.setCurrentIndex(2)
+        self.tabWidget_motor_stats.setCurrentIndex(3)
+        self.tabWidget_motor_stats.setCurrentIndex(4)
+        self.tabWidget_motor_stats.setCurrentIndex(5)
+        self.tabWidget_motor_stats.setCurrentIndex(0)
+
+        self._change_graphs(all=True, type="motor_stat", ncol=1)
+
+        self.t0 = 0
+        self.init_complete = True
+        # TODO: refresh graphs on resize?
+
+    def on_resize_main(self, empty):
+        if (self._widget.width() * self._widget.height()) < 3500000:
+            self.font_offset = -3
+        else:
+            self.font_offset = 1
+
+    def _include_tactile_plugin(self):
+        self.tactile_gui_list = []
+        self.timer = QTimer(self._widget)
+        number_of_biotacs = 5
```

Comment:
Probably better to put them in init as global variable:
self.number_of_biotacs = 5
self.electrodes_per_biotac = 25
and you can use them in all the functions. It will help with hands with less biotacs

Reply:
Done